### PR TITLE
Create and update content in the Apps Functionality topic

### PR DESCRIPTION
**Pull request by neflyte**
#### Summary
New and improved content for the Apps Functionality topic

#### Topics

- [x] Subscriptions
- [x] K/V store
- [x] External webhooks

### CODE REVIEW

File: site/content/integrate/apps/functionality/external-webhooks/_index.md
Comment (levb) [inline site/content/integrate/apps/functionality/external-webhooks/_index.md:46]: Need the webhook permission? See https://github.com/mattermost/mattermost-app-examples/blob/master/golang/webhooks/hello.go#L34

**Review by levb — APPROVED** [review]
Mostly LGTM, please add the reference to webhook permission.

File: site/content/integrate/apps/functionality/external-webhooks/_index.md
Comment (neflyte) [reply site/content/integrate/apps/functionality/external-webhooks/_index.md:47]: Great catch! Fixed. :)

File: site/content/integrate/apps/functionality/subscriptions.md
Comment (mickmister) [inline site/content/integrate/apps/functionality/subscriptions.md:32]: FYI We will need to revisit these with `bot_mentioned` etc. when we re-enable the post-related subscriptions

File: site/content/integrate/apps/functionality/subscriptions.md
Comment (mickmister) [inline site/content/integrate/apps/functionality/subscriptions.md:103]: Is the `Call` required when unsubscribing?

**Review by mickmister — APPROVED** [review]

File: site/content/integrate/apps/functionality/subscriptions.md
Comment (neflyte) [reply site/content/integrate/apps/functionality/subscriptions.md:32]: Please post a ticket/issue to document that change when it has been released. :)

Comment (cwarnermm) [reply site/content/integrate/apps/functionality/subscriptions.md:32]: We have a **Docs/Needed** label that can be applied to the PR when it's merged.

File: site/content/integrate/apps/functionality/subscriptions.md
Comment (neflyte) [reply site/content/integrate/apps/functionality/subscriptions.md]: After inspecting the code, it doesn't look that way, no. :) I've removed the Call field from the example.

**Review by mickmister — APPROVED** [review]
LGTM :+1: FYI @neflyte when the PR has been "approved" and any remaining changes to address are trivial, there's no need to re-request review from the reviewer. It's a bit subjective to determine this, but in this case, I think the change is small/simple enough that the previous approval still stands

**Review by levb — APPROVED** [review]

**Review by cwarnermm — APPROVED** [review]
Thanks, @neflyte!

**Pull request merged**
